### PR TITLE
""spatial hashing"" grid to optimize distance checking on clients for playsound

### PR DESCRIPTION
**Pull request by mybluecorners**
needs :
- testing to confirm it all works with multiple clients
- testing to confirm its actually faster than the default search

## Changelog

```
(u)CodeDude:
(*)Added soft soft pizza to the game - a tasty drink found in soda machines!
(+)Made some small sprite changes to the base pizza sprite.
```

### CODE REVIEW

File: code/modules/spatial_hashing.dm [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+		min_y = A.y - atomboundsize
+		max_x = A.x + atomboundsize
+		max_y = A.y + atomboundsize
+		ADD_BUCKET(min_x,A.y)
+		ADD_BUCKET(max_x,A.y)
+		ADD_BUCKET(A.x,min_y)
+		ADD_BUCKET(A.x,max_y)
+
+		//NW,NE,SW,SE
+		min_x = A.x - (atomboundsize * 0.7071)
```
Comment (ZeWaka) [inline code/modules/spatial_hashing.dm:98]: what is this constant

Comment (mybluecorners) [reply code/modules/spatial_hashing.dm:98]: lazy sqrt to get the 'edge' point of a diamond shape diagonally

Comment (ZeWaka) [reply code/modules/spatial_hashing.dm:98]: `(sqrt(2)/2)`

File: code/modules/spatial_hashing.dm [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+		min_y = A.y - (atomboundsize * 0.7071)
+		max_x = A.x + (atomboundsize * 0.7071)
+		max_y = A.y + (atomboundsize * 0.7071)
```
Comment (ZeWaka) [inline code/modules/spatial_hashing.dm:101]:
```suggestion
		min_x = A.x - (atomboundsize * (sqrt(2)/2))
		min_y = A.y - (atomboundsize * (sqrt(2)/2))
		max_x = A.x + (atomboundsize * (sqrt(2)/2))
		max_y = A.y + (atomboundsize * (sqrt(2)/2))
```

File: code/modules/spatial_hashing.dm (hunk 1 above)
Comment (Kyle2143) [reply code/modules/spatial_hashing.dm:102]: If performance is an issue and we're doing rounding on the result, a constant is going to be better than using the sqrt function.

Comment (ZeWaka) [reply code/modules/spatial_hashing.dm:102]: @Kyle2143 sqrt(constant) is done at compilation

**Pull request merged**